### ISSUE TITLE
Embedded document's property increasing from old value

### ISSUE THREAD

**Issue by mozkarakoc** [arangodb/arangodb#2002]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [x] 3.0
On this operating system:
- [x] MacOS, version: OS X El Capitan v 10.11.6

### this is an AQL-related issue:
I'm issuing AQL via:
- [x] web interface with this browser: Safari     running on this OS: MacOS, version: OS X El Capitan v 10.11.6

A document on my user collection as below: 
```json
{
    "_key": "123043079",
    "_id": "user/123043079",
    "_rev": "123045128",
    "createdAt": 1471538120000,
    "userName": "...",
    "email": "...",
    "password": "....",
    "follow": {
      "followee": 0,
      "follower": 0
    }
  }
```

I've run

```js
FOR u IN user 
UPDATE { _key: "123043079", follow: {
follower: 5 }}
 IN user
```
follower attribute updated to 5; but when I've run

```js
FOR u IN user 
UPDATE { _key: "123043079", follow: {
follower: u.follow.follower + 1}}
 IN user
```
follower attribute updated to 1 always. What's wrong? How can I increase the `followerCount` of my document?

**Comment by Simran-B** [comment]
Interesting... the query seems to work if there's a single document in the collection (here: `user`). If there's more than one, the results are somewhat confusing. Not sure if this is a loophole in the AQL syntax, will have to investigate further.

I think what happens is that you iterate over all documents in `user`, and for each document, the document with key `123043079` is updated - the same document several times. `u` refers to the current document and not always the document with key `123043079`. If `123043079` is the first document in the loop, its current value will be taken and incremented by 1. If there's another document in the collection, its value will be used, increased by 1, and assigned to `123043079`'s follower attribute. So the follower attribute is overwritten, with the value 1 if the other document's follower attribute is either 0, null or doesn't exist at all (will be implicitly be casted to 0, plus 1 = 1).

If you really just want to update that one single document, please use the following syntax:

```js
LET u = DOCUMENT("user/123043079")
UPDATE u WITH { follow: {
    follower: u.follow.follower + 1
} } IN user
```

Also possible, but slightly slower AFAIK:

```js
FOR u IN user
  FILTER u._key == "123043079"
  LIMIT 1
  UPDATE u WITH { follow: { follower: u.follow.follower + 1 } } IN user
```

**Comment by mozkarakoc** [comment]
Exactly as you said. There are many documents in collection and document with key "123043079" updated to last document's of collection in my query. 

Thx for your improvement about updating to single document.

**Comment by Simran-B** [comment]
BTW. this works as well to update a single document:

```js
UPDATE { _key: "123043079", follow: { follower: 123 } } IN user
```

The pseudo-variable `OLD` is not available inside the document to increment an existing attribute however. It would be handy if it was, and I suggested this internally as future improvement. Can't say how complicated it is to implement though.

For the time being, if you want to use this syntax, you would need to do this:

```js
LET doc = DOCUMENT("user/123043079")
UPDATE { _key: "123043079", follow: { follower: doc.follow.follower + 1 } } IN user
```

I like the `UPDATE ... WITH` variant I showed above better though.
